PullRequest for Issue22: refine the response message

XuetaoLiu-S opened the pull request:
- make validateResponse() function does validation work
- add function to print out the result
- remove the usage of qdebug, use AMDSErrorMonitor for all the debugging information

XuetaoLiu-S committed:
remove the usage of QDebug

dretrex commented on source/ClientRequest/AMDSClientRequest.h:
```diff
@@ -60,8 +60,12 @@ Q_OBJECT
 	/// validate the message response
 	virtual bool validateResponse() {return true;}
+	/// pure virtual to return a data string of the message
+	virtual QString toString() = 0;
```
Technically, a `toString()` method would be a const function since it isn't modifying any members of the class is represents.

dretrex commented:
Looks good other than my one comment.
Time Estimate: 1 minute

XuetaoLiu-S committed:
- Add the copy/clone function for the dataModels of AMDS, which is
needed when we need copy data
- For AMDSClientDataRequest, the data will be a copy of the original
data instead of the pointer to the original data
- when clearing the data in AMDSClientDataRequet, we will delete the
date since it is a copy now and safe to be deleted
- when doing flattening, we will delete the past +|/ result, since they
are newly created instances. In this way, we will avoid memory leak
- after reflattening, we will delete the array content since they are
newly "created" dataholder instance by the flatten function

XuetaoLiu-S committed:
Fix the bug for spectra data flattening

XuetaoLiu-S committed:
remove commented codes

XuetaoLiu-S committed:
Merge branch 'Issue67' into Issue73

XuetaoLiu-S committed:
make toString() function const

XuetaoLiu-S committed:
remove unused codes

XuetaoLiu-S committed:
fix potential bugs for the usage of deleteLater()

XuetaoLiu-S committed:
add getter function for lightWeightDataHolder of AMDSFullDataHolder class

dretrex commented:
Looks good.

XuetaoLiu-S committed:
Merge pull request #74 from acquaman/Issue73

PullRequest for Issue73: fix the bug for spectra data flatten

XuetaoLiu-S committed:
encrypt the usage of the local variables

XuetaoLiu-S committed:
Merge pull request #72 from acquaman/Issue67

PullRequest for Issue67: there are memory leak when doing data flattening

iainworkman commented:
Looks good.
Time Estimate: 10 minutes.

XuetaoLiu-S committed:
Fix cross class reference issues
